airflow/dag_processing/processor.py

```diff
@@ -647,3 +650,17 @@ def process_file(
             self.log.exception("Error logging import errors!")
 
         return len(dagbag.dags), len(dagbag.import_errors)
+
+    def _deactivate_missing_dags(self, session: Session, dagbag: DagBag, file_path: str) -> None:
+        for dag in (
+            session.query(DagModel)
+            .filter(DagModel.fileloc == file_path)
+            .filter(DagModel.is_active)
+            .filter(~DagModel.dag_id.in_(dagbag.dag_ids))
+            .all()
```

Comment:
You can chain this filter -- they use logical AND

```suggestion
            session.query(DagModel)
            .filter(DagModel.fileloc == file_path, DagModel.is_active, ~DagModel.dag_id.in_(dagbag.dag_ids))
            .all()
```